File: 백준/Silver/1920. 수 찾기/수 찾기.py
import sys
input = sys.stdin.readline

def binary_search(array, target, start, end):
    while start <= end:
        mid = (start + end)//2
        if array[mid] == target:
            return 1
        elif array[mid] > target:
            end = mid - 1
        else:
            start = mid + 1
    return 0

n = int(input())
array = list(map(int, input().split()))
m = int(input())
target = list(map(int, input().split()))

array.sort()
for i in target:
    print(binary_search(array, i, 0, len(array)-1 ))

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include<iostream>
#include<algorithm>
using namespace std;

int arr[100001];

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int n, m;
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> arr[i];
    }
    cin >> m;
    sort(arr, arr+n);
    for(int i = 0; i < m; i++) {
        int target;
        cin >> target;
        cout << binary_search(arr, arr+n, target) << "\n";
    }
}